Python/codecs.c

```diff
@@ -702,48 +702,47 @@ PyObject *PyCodec_IgnoreErrors(PyObject *exc)
 
 PyObject *PyCodec_ReplaceErrors(PyObject *exc)
 {
-    Py_ssize_t start, end, i, len;
+    Py_ssize_t start, end, slen;
 
     if (PyObject_TypeCheck(exc, (PyTypeObject *)PyExc_UnicodeEncodeError)) {
-        PyObject *res;
-        Py_UCS1 *outp;
-        if (PyUnicodeEncodeError_GetStart(exc, &start))
+        if (_PyUnicodeError_GetParams(exc, NULL, NULL,
+                                      &start, &end, &slen, false) < 0) {
             return NULL;
-        if (PyUnicodeEncodeError_GetEnd(exc, &end))
-            return NULL;
-        len = end - start;
-        res = PyUnicode_New(len, '?');
-        if (res == NULL)
+        }
+        PyObject *res = PyUnicode_New(slen, '?');
+        if (res == NULL) {
             return NULL;
+        }
         assert(PyUnicode_KIND(res) == PyUnicode_1BYTE_KIND);
-        outp = PyUnicode_1BYTE_DATA(res);
-        for (i = 0; i < len; ++i)
-            outp[i] = '?';
+        Py_UCS1 *outp = PyUnicode_1BYTE_DATA(res);
+        memset(outp, (int)'?', sizeof(Py_UCS1) * slen);
```

Comment:
While the cast is correct, I prefer to pass a `char` since it's a `Py_UCS1*` array:

```suggestion
        memset(outp, '?', sizeof(Py_UCS1) * slen);
```

Reply:
Actually, the signature of `memset` is `void * memset ( void * ptr, int value, size_t num );`, so the cast was for calling `memset`. But the compiler won't complain I think (I don't have a warning on my side but I don't know for Windows)

Reply:
Windows seems happy so I'll keep the cast removed.